### PR TITLE
build: move features dependencies from the main Cargo.toml file to th…

### PR DESCRIPTION
**Pull request by Yael-Starkware**
…e local Cargo.toml files

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Build-related changes

Short summary regarding the features dependencies:
I tried to remove entirely (from all toml files) each feature that appears in the main Cargo.toml file and check of any test failed. Tested both from the main dir and the local crate dirs, with cargo clean; rm Crago.lock in between, and these are the results:
1. hyper, features "full" - no test failed
2. insta, features "yaml" - could not compile crates/papyrus_storage/src/serializers_test.rs- could not find `assert_yaml_snapshot` in `insta`
3. schemrs , features "preserve_order" - no test failed
4. primitive-types. features "serde" - no test failed
TL;DR: The test don't cover these feature dependencies.

## Does this introduce a breaking change?

- [ ] No

## Other information

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/977)

### CODE REVIEW

**Review by yair-starkware — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/977#-:-NbDqHoU5Ql6YSUMZ-Rz:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @dan-starkware)

**Comment by yair-starkware** [comment]
Checking again

**Review by yair-starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/977)* status: all files reviewed, 2 unresolved discussions (waiting on @dan-starkware and @Yael-Starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/papyrus/977#-NbERaq19bfFhFojkm2-:-NbERaq19bfFhFojkm20:banocy3) (no related file):*
After discussions, we decided to consider missing the "arbitrary_precision" feature as a bug and add a test that checks that the feature is activated in all of the crates that use serde_json.

Please add the test called `serialization_precision` to all of the crates.

___

**Review by Yael-Starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/977)* status: 0 of 22 files reviewed, 2 unresolved discussions (waiting on @dan-starkware and @yair-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/papyrus/977#-NbERaq19bfFhFojkm2-:-NbTT5flBSxuSKmHlk8e:bs7wumb) (no related file):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

After discussions, we decided to consider missing the "arbitrary_precision" feature as a bug and add a test that checks that the feature is activated in all of the crates that use serde_json.

Please add the test called `serialization_precision` to all of the crates.
</blockquote></details>

Done.

___
___
*[`crates/papyrus_storage/Cargo.toml` line 41 at r1](https://reviewable.io/reviews/starkware-libs/papyrus/977#-NbECGIMCLwHqH5bxyQi:-NbTTchd7y0SZ9B9Yc0d:b-ev1iw4) ([raw file](https://github.com/starkware-libs/papyrus/blob/2e431ad07a98a77895c7119dd67f83cc33434f94/crates/papyrus_storage/Cargo.toml#L41)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

Add json feature
</blockquote></details>

as we discussed, I checked with cargo tree and the json feature in not coming from any other dependency. So I believe it is redundant to add it here.

**Review by yair-starkware — CHANGES_REQUESTED** [review]
> Reviewed 17 of 21 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/977)* status: 21 of 22 files reviewed, 4 unresolved discussions (waiting on @dan-starkware and @Yael-Starkware)

___
*[`crates/papyrus_gateway/src/v0_3_0/api/test.rs` line 1828 at r3](https://reviewable.io/reviews/starkware-libs/papyrus/977#-Nbhj3WE3MIX1dWRFeY4:-Nbhj3WE3MIX1dWRFeY5:b-xiqxrj) ([raw file](https://github.com/starkware-libs/papyrus/blob/9341967a3a78a6e2a4afc5475c9e5327f90ffa83/crates/papyrus_gateway/src/v0_3_0/api/test.rs#L1828)):*
> ```XML
> }
> 
> #[test]
> ```

Move the test to gateway_test and remove from the versions tests
___
*[`crates/papyrus_gateway/src/v0_4_0/api/test.rs` line 1817 at r3](https://reviewable.io/reviews/starkware-libs/papyrus/977#-NbhjNy1-AGQpO63n7YD:-NbhjNy1-AGQpO63n7YE:b-svyrie) ([raw file](https://github.com/starkware-libs/papyrus/blob/9341967a3a78a6e2a4afc5475c9e5327f90ffa83/crates/papyrus_gateway/src/v0_4_0/api/test.rs#L1817)):*
> ```XML
>         .unwrap();
> }
> ```

same (we need to test this once per crate)
___
*[`crates/papyrus_load_test/Cargo.toml` line 13 at r3](https://reviewable.io/reviews/starkware-libs/papyrus/977#-NbhjeIP7a9nk6n5vIZy:-NbhjeIP7a9nk6n5vIZz:b-nl9b5w) ([raw file](https://github.com/starkware-libs/papyrus/blob/9341967a3a78a6e2a4afc5475c9e5327f90ffa83/crates/papyrus_load_test/Cargo.toml#L13)):*
> ```TOML
> goose = "0.17.0"
> once_cell.workspace = true
> pretty_assertions.workspace = true
> ```

Move to dev-dependencies



_Code quote:_
```TOML
pretty_assertions.workspace = true
```
___
*[`crates/test_utils/Cargo.toml` line 16 at r3](https://reviewable.io/reviews/starkware-libs/papyrus/977#-Nbhk3iZ764hFnI67wlE:-Nbhk3iZ764hFnI67wlF:bg6z6qu) ([raw file](https://github.com/starkware-libs/papyrus/blob/9341967a3a78a6e2a4afc5475c9e5327f90ffa83/crates/test_utils/Cargo.toml#L16)):*
> ```TOML
> cairo-lang-utils.workspace = true
> indexmap = { workspace = true, features = ["serde"] }
> pretty_assertions.workspace = true
> ```

dev-dependencies



_Code quote:_
```TOML
pretty_assertions.workspace = true
```

**Review by Yael-Starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/977)* status: 17 of 23 files reviewed, 3 unresolved discussions (waiting on @dan-starkware and @yair-starkware)

___
*[`crates/papyrus_load_test/Cargo.toml` line 13 at r3](https://reviewable.io/reviews/starkware-libs/papyrus/977#-NbhjeIP7a9nk6n5vIZy:-NbhxWNv0-Pdtc_p0JlT:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/9341967a3a78a6e2a4afc5475c9e5327f90ffa83/crates/papyrus_load_test/Cargo.toml#L13)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

Move to dev-dependencies
</blockquote></details>

Done.
___
*[`crates/test_utils/Cargo.toml` line 16 at r3](https://reviewable.io/reviews/starkware-libs/papyrus/977#-Nbhk3iZ764hFnI67wlE:-NbhxXKt6cvrdBgf5ip5:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/9341967a3a78a6e2a4afc5475c9e5327f90ffa83/crates/test_utils/Cargo.toml#L16)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

dev-dependencies
</blockquote></details>

Done.
___
*[`crates/papyrus_gateway/src/v0_3_0/api/test.rs` line 1828 at r3](https://reviewable.io/reviews/starkware-libs/papyrus/977#-Nbhj3WE3MIX1dWRFeY4:-NbhxVQ0Dq7FUFukAChb:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/9341967a3a78a6e2a4afc5475c9e5327f90ffa83/crates/papyrus_gateway/src/v0_3_0/api/test.rs#L1828)):*
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

Move the test to gateway_test and remove from the versions tests
</blockquote></details>

Done.

**Review by yair-starkware — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/977#-:-Nbhz4v73Zz42f-gkkAq:bnfp4nl)* status: 22 of 23 files reviewed, all discussions resolved (waiting on @dan-starkware)
